public/app/types/unified-alerting-dto.ts

```diff
@@ -92,16 +92,12 @@ export enum GrafanaAlertState {
   KeepLastState = 'KeepLastState',
   OK = 'OK',
 }
-
-export interface GrafanaQueryModel extends DataQuery {
-  datasource: string;
-  datasourceUid: string;
-}
 export interface GrafanaQuery {
   refId: string;
   queryType: string;
   relativeTimeRange: RelativeTimeRange;
-  model: GrafanaQueryModel;
+  datasourceUid: string;
+  model: DataQuery;
```

Comment:
```suggestion
  model: DataQuery | ExpressionQuery;
```
?

Reply:
I don't think we need to do this since the `ExpressionQuery` is a `DataQuery`.

Reply:
but we won't have access to `ExpressionQuery` props when doing something like `model.expression` ?

Reply:
I think we have to do a guard check to know what type of model we have. But that needs to be done regardless of the `model: DataQuery` or `model: DataQuery | ExpressionQuery` if I'm not thinking totally wrong.